### PR TITLE
Added sms to All, still working on sms to dafaulters

### PR DESCRIPTION
**Pull request by raoaliahmad**
Check It

### CODE REVIEW

**Comment by raoaliahmad** [comment]
Added all sms functionalities except Sms to fee dafulters

File: web/src/modules/SMS/SmsOptions/ToSingleTeacher.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react'
+import { smsIntentLink } from 'utils/intent'
+
+import '../../style.css'
```
Comment (taimur38) [inline web/src/modules/SMS/SmsOptions/ToSingleTeacher.js]: default styles are imported across the app already. we only ever import styles when we have custom css for the component

File: web/src/modules/SMS/SmsOptions/ToSingleTeacher.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	return (
+	<div>
+		<div className="row">
+			<label>Select Teacher</label>		
+			<select {...this.former.super_handle(["selected_teacher_number"])}>
+				{
+					[<option key="abcd" value="" disabled>Select a Teacher</option>,
+					...Object.entries(teachers)
+					.filter(([id, teacher]) => teacher.Phone !== undefined && teacher.Phone !== "")
+					.map(([id, teacher]) => <option key={id} value={teacher.Phone}>{teacher.Name}</option>)
```
Comment (taimur38) [inline web/src/modules/SMS/SmsOptions/ToSingleTeacher.js]: indenty

File: web/src/modules/SMS/index.js [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
-import {getSectionsFromClasses} from 'utils/getSectionsFromClasses';
 import former from 'utils/former'
 import Layout from 'components/Layout'
 import Banner from 'components/Banner'
 
+import ToSingleStudent from 'SmsOptions';
+import ToSingleClass   from 'SmsOptions';
+import ToAllStudents   from 'SmsOptions';
+import ToSingleTeacher from 'SmsOptions';
+import ToAllTeachers   from 'SmsOptions';
```
Comment (taimur38) [inline web/src/modules/SMS/index.js]: make sure to make this a relative import './SmsOptions'

File: web/src/modules/SMS/index.js [hunk 4]
```diff
@@ -56,75 +58,64 @@ class SMS extends Component {
 		setTimeout(() => this.setState({ banner: {active: false}}), 3000);
 	}
 
-	sendMessage = () => {
+	sendMessage = (text, number) => {
```
Comment (taimur38) [inline web/src/modules/SMS/index.js]: Make sure to make a sendBatchMessages method and pass that through to the children,

File: web/src/modules/SMS/SmsOptions/ToFeeDefaulters.js [hunk 5]
```diff
@@ -11,22 +8,20 @@ export default class ToFeeDefaulters extends Component {
...
+	const { students, sendBatchMessages } = this.props;
 	
-	const message = { messages : [ students.filter(student => {student.Phone !== undefined && student.Phone !== ""})
-						.map(S => [{ number: S.Phone, text : this.state.text }])
-					]
-				}
+	const messages = Object.values(students)
+	.filter(student => Object.values(student.payments)
+	.reduce((agg, curr) => agg - (curr.type === "SUBMITTED" || curr.type === "FORGIVEN" ? 1 : -1) * curr.amount, 0) > 0 && student.Phone!== undefined && student.Phone !== "" )
+	.map(S => ({ number: S.Phone, text : this.state.text }));
```
[the remark this answers is not in the document]
Comment (raoaliahmad) [reply web/src/modules/SMS/SmsOptions/ToFeeDefaulters.js]: I'm calculating the total dues of a student and then checking if it's ( > 0)  i.e they owe us so student is a fee defaulter

[the remark this answers is not in the document]
Comment (taimur38) [reply web/src/modules/SMS/SmsOptions/ToFeeDefaulters.js]: looks good actually just fix the indentation

**Pull request merged**